specification/quantumleap.yml

```diff
@@ -0,0 +1,500 @@
+    in: path
+    required: true
+    name: entityType
+    type: string
+    description: "The entity type in which the query is restricted on. Used to constraint the search to entities of this type, specially useful when the entity id is not enough to uniquely identify an entity."
+    # example: Room  # Only Valid in OAPI 3.0
+  entityIdInPath:
+    in: path
+    required: true
+    name: entityId
+    type: string
+    description: "The entity id in which the query is restricted on. If the id is unique among all entity types, this could be used to uniquely identify the entity instance. Otherwise, you will have to use the entityType attribute to resolve ambiguity."
+    # example: Kitchen1  # Only Valid in OAPI 3.0
+  attrNameInPath:
+    in: path
+    required: true
+    name: attrName
+    type: string
+    description: "The name of the attribute the query is interested on. It must be a valid attribute of all the entities covered by the query scope."
+    # example: temperature  # Only Valid in OAPI 3.0
+
+################################################################################
+# PARAMETERS: IN QUERY
+################################################################################
+  type:
+    in: query
+    name: type
+    type: string
+    description: "Comma-separated list of entity types whose data are to be included in the response. Use only one (no comma) when required. If used to resolve ambiguity for the given entityId, make sure the given entityId exists for this entityType."
+  id:
+    in: query
+    name: id
+    type: string
+    description: "Comma-separated list of entity ids whose data are to be included in the response."
+  idPattern:
+    in: query
+    name: idPattern
+    type: string
+    description: "A correctly formated regular expression. Retrieve entities whose ID matches the regular expression. Incompatible with id."
+  attrs:
+    in: query
+    name: attrs
+    type: string
+    description: "Comma-separated list of attribute names whose data are to be included in the response."
+  aggrMethod:
+    in: query
+    name: aggrMethod
+    type: string
+    enum: [count, sum, avg, min, max]  # These taken from ISO standard on SQL.
+    # Remember Crate supports many others https://crate.io/docs/crate/reference/en/latest/sql/aggregation.html
+    description: "Optional. The function to apply to the raw data filtered by the query parameters. If not given, the returned data are the same raw inserted data."
+  aggrPeriod:
+    in: query
+    name: aggrPeriod
+    type: string
+    enum: [year, month, day, hour, minute, second]
+    default: month
+    description: "Optional. If defined, the aggregation function will be applied N times, once for each period, and all those results will be considered for the result. For example, a query asking for the average temperature of an attribute will typically return 1 value. With an aggregationPeriod of day, you get the daily average of the temperature instead."
```

Comment:
if not the defined, shouldn't it be on the overall data retrieved?

Reply:
I'm afraid I don't understand the question, could you rephrase?

Reply:
let's suppose that i don't specific an aggregation period. the aggregate should be compute on the overall time frame retrieved.